File: src/005.LinkedList/083.RemoveDuplicatesFromSortedList/83.删除排序链表中的重复元素.go
/*
 * @lc app=leetcode.cn id=83 lang=golang
 *
 * [83] 删除排序链表中的重复元素
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 // 记录点: 因为链表是有序, 所有只需要判断当前元素和下一个元素是否相同即可, 不需要单独 map 进行记录
func deleteDuplicates(head *ListNode) *ListNode {
	if head == nil {
		return nil
	}

	cur := head

	for cur.Next != nil {
		if cur.Val == cur.Next.Val {
			cur.Next = cur.Next.Next
		} else {
			cur = cur.Next
		}
	}

	return head
}
// @lc code=end

File: src/005.LinkedList/083.RemoveDuplicatesFromSortedList/83.删除排序链表中的重复元素.js
/*
 * @lc app=leetcode.cn id=83 lang=javascript
 *
 * [83] 删除排序链表中的重复元素
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
  if (!head) {
    return head
  }

  const countMap = {}
  let list = head

  countMap[list.val] = 1

  while (list.next) {
    if (countMap[list.next.val]) {
      list.next = list.next && list.next.next
    } else {
      countMap[list.next.val] = 1
      list = list.next
    }
  }

  return head
};
// @lc code=end

